Initial skeleton of design

EwenPinto opened the pull request:
/_Initial design of yadl containing skeleton of API's to be included in yadl_/

Signed-off-by: Ewen Pinto ewenpin@gmail.com

josephaug26 committed:
Initial skeleton of design
/*Initial design of yadl containing skeleton of API's to be included in yadl*/

Signed-off-by: Ewen Pinto <ewenpin@gmail.com>

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+{
+	O_APPEND = 1.
+	O_RDONLY,
+	O_WRONLY
```
You are missing Read Write mode.

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+
+struct ydlobject
+{
+    	uint64/uuid dobject_id;
```
Will this compile!!!??

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+typedef enum ydlerrors 
```
Comments on what this enum means? Same with other structures/enums

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+	ydl_stderr = -1;     
+}ydl_error_t;
+
+enum hash_type
```
This should come from hash.h

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+	md5
+};
+
+enum chunk_type
```
This should come from chunk.h

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+    	variable_chunk
+};
+
+struct ydl_store_config
```
should come from store.h

josephaug26 replied:
pick the enums from the existing yadl code and dont re-invent for example .

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+    	struct ydlkey_value *next;
+};
+
+struct ydlobject
```
rename this to ydl_object_t

josephaug26 commented on design/yadl.h:
```diff
@@ -0,0 +1,118 @@
+    	ydlobject_meta *;
+};
+
+enum ydl_mode
```
rename this to ydl_mode_t and please follow the same convention everywhere

josephaug26 commented on design/yadl.c:
```diff
@@ -0,0 +1,81 @@
+#include "yadl.h"
+
+ydlobject  
+*ydlopen(ydlobject_meta *, ydl_mode, enum ydlerror,int *return_value) 
```
WIll this code compile!!!!???

Commit:
Merge pull request #18 from gitter-badger/gitter-badge

Add a Gitter chat badge to README.md

EwenPinto committed:
Merge branch 'design' of github.com:YADL/yadl into design

EwenPinto committed:
Incremental patch for yadl design
/*Fixed all the review comments
Implemented initial skeleton of Design in yadl.c and yadl.h*/
Signed-off-by: Ewen Pinto <ewenpin@gmail.com>
